providers/amazon/tests/system/amazon/aws/example_sagemaker.py

```diff
@@ -107,6 +108,39 @@ def main():
 """)
 
 
+def _install_aws_cli_if_needed():
+    """
+    Check if the AWS CLI tool is installed and install it if needed.
+
+    The AmazonLinux image has flip-flopped a couple of times on whether this is included in the base image
+    or not, so to future-proof this we are going to check if it's installed and install if necessary.
```

Comment:
Was it our build infra that was actually ensuring this was installed? And perhaps that's regressed? I think this is fine either way, but it's worth looking into.